src/KubernetesClient/CertUtils.cs

```diff
@@ -19,22 +18,32 @@ public static class CertUtils
         /// Load pem encoded cert file
         /// </summary>
         /// <param name="file">Path to pem encoded cert file</param>
-        /// <returns>x509 instance.</returns>
-        public static X509Certificate2 LoadPemFileCert(string file)
+        /// <returns>List of x509 instances.</returns>
+        public static IList<X509Certificate2> LoadPemFileCert(string file)
         {
-            var certs = new X509CertificateParser().ReadCertificates(File.OpenRead(file));
-            var store = new Pkcs12StoreBuilder().Build();
-            foreach (X509Certificate cert in certs)
-            {
-                store.SetCertificateEntry(Guid.NewGuid().ToString(), new X509CertificateEntry(cert));
-            }
+            var certs = new List<X509Certificate2>();
 
-            using (var pkcs = new MemoryStream())
+            var certdata = File.ReadAllText(file)
+                .Replace("\r", "")
+                .Replace("\n", "");
+
+            // Retrieve all certificates from the file
+            //
+            var r = new Regex("-----BEGIN CERTIFICATE-----(.*?)-----END CERTIFICATE-----");
+
+            var matches = r.Matches(certdata);
```

Comment:
suggest to be `new X509CertificateParser().ReadCertificates` instead of building it ourselves.

Reply:
Definitely better. Only caveat is that we need to do some additional steps to convert the bouncy x509 certificates to .Net crypt x509 certs to store them in an X509Certificate2Collection, but that is trivial.